### PR TITLE
[cdc] Use CdcSourceRecord as source type instead of raw String

### PR DESCRIPTION
**Pull request by harveyyue**
### Purpose

Refactor the cdc source record, it will help to implement more data formats.

### CODE REVIEW

**Comment by harveyyue** [comment]
@yuzelin Could you help to review it?

**Comment by yuzelin** [comment]
Thanks for contributing! We are planning to cut release 0.7.0 branch today (Jan. 25). Since this PR is refactor, I will review later.

**Comment by zhuangchong** [comment]
I have optimized a version, and next I want to implement records that receive generic structures, but this is very difficult to modify.

File: paimon-flink/paimon-flink-cdc/src/main/java/org/apache/paimon/flink/action/cdc/CdcSourceRecord.java [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+
+import org.apache.paimon.annotation.Experimental;
+
+import javax.annotation.Nullable;
+
+import java.io.Serializable;
+import java.util.Objects;
+
+/** A data change record from the CDC source. */
+@Experimental
```
Comment (yuzelin) [inline paimon-flink/paimon-flink-cdc/src/main/java/org/apache/paimon/flink/action/cdc/CdcSourceRecord.java:29]: I think it's just an internal implementation, so we don't need to annotate 'Experimental'.

File: paimon-flink/paimon-flink-cdc/src/main/java/org/apache/paimon/flink/action/cdc/format/RecordParser.java [hunk 2]
```diff
@@ -86,26 +89,21 @@ public RecordParser(
...
-            if (!recordOpt.isPresent()) {
-                return null;
-            }
-
-            Schema.Builder builder = Schema.newBuilder();
-            recordOpt.get().fieldTypes().forEach(builder::column);
-            builder.primaryKey(extractPrimaryKeys());
-            return builder.build();
-        } catch (Exception e) {
-            logInvalidJsonString(record);
```
Comment (yuzelin) [inline paimon-flink/paimon-flink-cdc/src/main/java/org/apache/paimon/flink/action/cdc/format/RecordParser.java:106]: I think it's better to keep this. We catch all exception to log the original record for trouble shooting.

File: paimon-flink/paimon-flink-cdc/src/main/java/org/apache/paimon/flink/action/cdc/format/RecordParser.java [hunk 3]
```diff
@@ -126,21 +124,16 @@ protected LinkedHashMap<String, DataType> fillDefaultTypes(JsonNode record) {
     }
 
     @Override
-    public void flatMap(String value, Collector<RichCdcMultiplexRecord> out) {
-        try {
-            setRoot(value);
-            extractRecords().forEach(out::collect);
-        } catch (Exception e) {
-            logInvalidJsonString(value);
```
Comment (yuzelin) [inline paimon-flink/paimon-flink-cdc/src/main/java/org/apache/paimon/flink/action/cdc/format/RecordParser.java:134]: Same as above.

**Comment by yuzelin** [comment]
Thanks @harveyyue and sorry for my late response. Thanks @zhuangchong for resolving the conflicts and tests!

**Review by yuzelin — APPROVED** [review]
+1

**Pull request merged**